packages/expo-dev-launcher/plugin/src/withDevLauncherAppDelegate.ts

```diff
@@ -106,30 +207,61 @@ export function modifyAppDelegate(appDelegate: string, expoUpdatesVersion: strin
     );
   }
 
-  if (!appDelegate.includes(DEV_LAUNCHER_APP_DELEGATE_SOURCE_FOR_URL)) {
-    appDelegate = appDelegate.replace(
-      'return [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index" fallbackResource:nil];',
-      DEV_LAUNCHER_APP_DELEGATE_SOURCE_FOR_URL
-    );
+  appDelegate = changeDebugURL(appDelegate);
+  appDelegate = addDeepLinkHandler(appDelegate);
+
+  if (!appDelegate.includes(DEV_LAUNCHER_APP_DELEGATE_CONTROLLER_DELEGATE_LEGACY)) {
+    appDelegate += DEV_LAUNCHER_APP_DELEGATE_CONTROLLER_DELEGATE_LEGACY;
   }
 
-  if (!appDelegate.includes(DEV_LAUNCHER_APP_DELEGATE_ON_DEEP_LINK)) {
+  appDelegate = removeDevMenuInit(appDelegate);
+  return appDelegate;
+}
+
+export function modifyAppDelegate(appDelegate: string, expoUpdatesVersion: string | null = null) {
+  const shouldAddUpdatesIntegration =
+    expoUpdatesVersion != null && semver.gt(expoUpdatesVersion, '0.6.0');
+
+  if (!appDelegate.includes(DEV_LAUNCHER_INITIALIZE_REACT_NATIVE_APP_FUNCTION_DEFINITION)) {
```

Comment:
this seems like a really brittle check -- any reason we couldn't use `NEW_INITIALIZE_REACT_NATIVE_APP_FUNCTION` instead?